sql/catalyst/src/main/scala/org/apache/spark/sql/connector/catalog/LookupCatalog.scala

```diff
@@ -92,14 +92,14 @@ private[sql] trait LookupCatalog extends Logging {
    * Catalog name takes precedence over namespaces.
    */
   object CatalogAndNamespace {
-    def unapply(parts: Seq[String]): Some[(CatalogPlugin, Option[Seq[String]])] = parts match {
+    def unapply(parts: Seq[String]): Some[(CatalogPlugin, Seq[String])] = parts match {
```

Comment:
nit: shall we make `CatalogAndNamespace` and `CatalogAndTable` consistent? e.g. use `nameParts` as parameter name, and have `assert(nameParts.nonEmpty)` at the begining.

Reply:
the class doc should be consistent too